src/Data/Version.php

```diff
@@ -0,0 +1,94 @@
+<?php declare(strict_types=1);
+
+namespace ILIAS\Data;
+
+/**
+ * A version number that consists of three numbers (major, minor, patch).
+ */
+class Version
+{
+    private const REGEXP = '(\d+)([.](\d+)([.](\d+))?)?';
+
+    protected int $major;
+    protected int $minor;
+    protected int $patch;
+
+    public function __construct(string $version)
+    {
+        $match = [];
+        if (!preg_match("/" . self::REGEXP . "/", $version, $match)) {
+            throw new \InvalidArgumentException(
+                "Expected version string to match this regular expression: " . self::REGEXP
+            );
+        }
+        $this->major = (int) $match[1];
+        $this->minor = 0;
+        $this->patch = 0;
+        if (count($match) > 3) {
+            $this->minor = (int) $match[3];
+        }
+        if (count($match) > 5) {
+            $this->patch = (int) $match[5];
+        }
+    }
+
+    public function getMajor() : int
+    {
+        return $this->major;
+    }
+
+    public function getMinor() : int
+    {
+        return $this->minor;
+    }
+
+    public function getPatch() : int
+    {
+        return $this->patch;
+    }
+
+    public function equals(Version $other) : bool
+    {
+        return
+            $this->major === $other->major
+            && $this->minor === $other->minor
+            && $this->patch === $other->patch;
+    }
+
+    public function isGreaterThan(Version $other) : bool
+    {
+        if ($this->major > $other->major) {
```

Comment:
You could use the `__toString` method (or `(string` cast) of both values and use `version_compare()` instead.

Reply:
I'll leave it like this to be more explicit about the algorithm. I somehow expect that we will want to add more parts to the version at some point. Thanks anyway for the suggestion!